shadowfinder/shadowfinder.py

```diff
@@ -138,28 +164,44 @@ def find_shadows(self):
 
         valid_sun_altitudes = pos_obj["altitude"]  # in radians
 
-        # Calculate the shadow length
-        shadow_lengths = self.object_height / np.apply_along_axis(
-            np.tan, 0, valid_sun_altitudes
-        )
+        if self.sun_altitude_angle is not None:
```

Comment:
Would you add comments to the code explaining each section of the if else.

It may even be good to have it as if... elif... else: error, in case a situation arises where the state of the object gets screwed up somehow.

Reply:
Not sure if I added to much text now, and if its even needed.